recipes/zserio/all/conanfile.py

```diff
@@ -50,13 +63,21 @@ def validate(self):
         if self.settings.compiler.cppstd:
             check_min_cppstd(self, self._min_cppstd)
 
+        minimum_compiler_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_compiler_version and Version(self.settings.compiler.version) < minimum_compiler_version:
+            raise ConanInvalidConfiguration(
+                f"Compiler version '{self.settings.compiler.version}' not supported, "
+                f"minumum is '{minimum_compiler_version}'!"
+            )
+
     def source(self):
         sources = self.conan_data["sources"][self.version]
         get(self, **sources["runtime"], strip_root=True)
         download(self, filename="LICENSE", **sources["license"])
 
     def generate(self):
         tc = CMakeToolchain(self)
+        tc.variables["CMAKE_CXX_STANDARD"] = self._min_cppstd
```

Comment:
Thanks for the quick response! There were some compilers in the pipelines which don't have C++11 enabled by default and the build didn't work - see build logs for [MacOS Clang 13](https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/pr/23846/2-macos-m1-clang/zserio/2.14.0//summary.json) and [gcc 5](https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/pr/23846/1-linux-gcc-legacy/zserio/2.14.0//summary.json).
I just didn't know how to fix it and I've found similar solution in another packages.

The only variant I was thinking about was to add:
```python
if not valid_min_cppstd(self, self._min_cppstd):
    tc.variables["CMAKE_CXX_STANDARD"] = self._min_cppstd
```
which would probably allow to set higher C++ standard. However the Zserio C++ generator states that it's C++11, so I decided to force the standard to C++11. But I would probably accept any solution which will ensure that at least C++11 will be used.

Reply:
Oh, I see! This is one of the painpoints we hope to alleviate once we drop support for Conan 1 in the pipeline, but for now we need to take care of these situations.

If you check the "Contributors are allowed to push to this branch" in the righthand pane in the PR, I'll be able to directly push a fix for you for this case :)

Reply:
> Contributors are allowed to push to this branch

I don't see such an option, the PR is created from a fork, maybe it's the reason? I think you will have to create a PR on my branch? https://github.com/ndsev/zserio-conan-center-index/tree/zserio-2.14.0

Reply:
Weird, I had supposed that https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests would show up at all times!

Reply:
Maybe it's related to this [issue](https://github.com/isaacs/github/issues/1681)? Also in a [discussion](https://github.com/orgs/community/discussions/5634).
![image](https://github.com/conan-io/conan-center-index/assets/8450210/aa231a58-5b8e-4906-837e-2e9b7b10b605)

Reply:
TIL, thanks for investigating it :) I'll open a pr on your fork then, sorry for the noise!

Reply:
> However the Zserio C++ generator states that it's C++11, so I decided to force the standard to C++11.

I would suggest opening an issue to the upstream asking to fix it. The upstream would expect a modern compiler, that's not the first time we such case. It could be fixed by including `target_compiler_features(<target> PUBLIC cxx_std_11)` in their CMake file.

Reply:
> would impose minimal C++11 standard as required

Aha, I didn't know that, thanks, we will investigate it the related issue... However the Zserio release 2.14.0 is already published...